spec/system/search_spec.rb

```diff
@@ -270,7 +270,7 @@
     #  browse mode is bare link
     visit "/"
     find("#general-search-button").click
-    expect(page).to have_link href: "/search.csv"
+    expect(page).to have_link href: "search.csv?search=&starting_year=&ending_year="
```

Comment:
If there is a transient failure here, it's unclear to me why it's occurring thus how this change resolves the issue. Does it resolve the issue, or is it making the issue less likely to occur?

Reply:
I think I've figured it out. When the page loads for the first time, the link with `href="/search.csv"` exists. When the search button is clicked in line 272, some js changes the csv link to match the current search (in this case an empty search).  The old line 273 would look for `href="/search.csv"` and it would sometimes find it, before the js would change the it. Most of the time it wouldn't though.

Capybara has an [implicit wait ](https://github.com/teamcapybara/capybara#asynchronous-javascript-ajax-and-friends) of two seconds, if it can't find what it is looking for at first. So it will wait for two seconds for the `href: "/search.csv"` to change to `href: "search.csv?search=&starting_year=&ending_year="`

The update line 273 matches what we want it to be after the js changes it. Capybara will consistently wait for it to change now.